### PR TITLE
Update contract CI to ensure docs & types are built

### PR DESCRIPTION
**Pull request by hstove**
This PR does two things:

- Updates the `pnpm build` command in `contracts` to build Clarigen docs & types. This is implicitly included in `make build`.
- Adds a CI step to check that docs & types were built and pushed

The motivation is that it's easy to forget to run `clarigen && clarigen docs` before pushing, and doing so can introduce git changes in other branches, or even end up with type or runtime errors.

You can see the CI in action:

- The first commit only adds the CI workflow update, but doesn't include updated types. [CI fails and an error is shown](https://github.com/stacks-network/sbtc/actions/runs/9130353433/job/25107067998)
- The second commit pushes the types that were created with `make build`, and CI passes.

### CODE REVIEW

**Comment by hstove** [comment]
NB: this builds on top of `signer-setup` because that includes Clarity changes that change the types/docs, so I'm using that to demonstrate the CI. We should wait until https://github.com/stacks-network/sbtc/pull/151 is merged (it's approved), and then switch this to be merged into `main`, and then merge this PR once approved.

**Review by setzeus — APPROVED** [review]
Mainly documentation & tooling work, LGTM! Thank you for improving the process.

**Pull request merged**
